hermeznetwork/bridge-ui
# Add integration with a fiat exchange rates API

### What does this PR does?

This PR adds the integration with the https://exchangeratesapi.io service, so we are able to convert a token price from USD to the user preferred currency.

### How to test?

To test this feature you can write this code in the `home.view.tsx` file and change the currency in the `Settings` view of the UI. After going back to the home, the value logged in the console should change accordingly.

```ts
const { preferredCurrency, getTokenPrice } = usePriceOracleContext();

useEffect(() => {
  if (preferredCurrency) {
    getTokenPrice("0x7D1AfA7B718fb893dB30A3aBc0Cfc608AaCfeBB0", preferredCurrency)
      .then((tokenPrice) => {
        console.log(tokenPrice);
      })
      .catch((err) => console.log(err));
  }
}, [preferredCurrency, getTokenPrice]);
```

## Checklist

These are the criteria that every PR should meet, please check them off as you
review them:
